### ISSUE TITLE
UnhandledPromiseRejectionWarning: RangeError: Invalid string length

### ISSUE THREAD

**Issue by angulardev123** [webpack/webpack-cli#2762]
**Describe the bug**

I have a large enterprise level Angular 12 application with hundreds of components.  I'm getting the following error during the production build.  The output below is from Webpack 5 but this is happening in Webpack 4 as well.

**What is the current behavior?**

```
C:\myapps\app1>node --max_old_space_size=16384 .\node_modules\webpack\bin\webpack.js --color --config webpack.prod.js
(node:18764) [DEP_WEBPACK_RULE_LOADER_OPTIONS_STRING] DeprecationWarning: Using a string as loader options is deprecated (ruleSet[1].rules[1].use[0].options)
(node:18764) UnhandledPromiseRejectionWarning: RangeError: Invalid string length
    at Array.join (<anonymous>)
    at itemsJoinMoreSpacing (C:\myapps\app1\node_modules\webpack\lib\stats\DefaultStatsPrinterPlugin.js:791:61)
    at Hook.eval [as call] (eval at create (C:\myapps\app1\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:5:16)
    at Hook.CALL_DELEGATE [as _call] (C:\myapps\app1\node_modules\tapable\lib\Hook.js:14:14)
    at C:\myapps\app1\node_modules\webpack\lib\stats\StatsPrinter.js:209:8
    at StatsPrinter._forEachLevel (C:\myapps\app1\node_modules\webpack\lib\stats\StatsPrinter.js:126:19)
    at StatsPrinter._print (C:\myapps\app1\node_modules\webpack\lib\stats\StatsPrinter.js:208:24)
    at StatsPrinter.print (C:\myapps\app1\node_modules\webpack\lib\stats\StatsPrinter.js:156:16)
    at C:\myapps\app1\node_modules\webpack\lib\stats\StatsPrinter.js:223:27
    at Array.map (<anonymous>)
(node:18764) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:18764) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

The issue seems to be in line 791 of DefaultStatsPrinterPlugin.js during the join call.

![webpackerror](https://user-images.githubusercontent.com/85368155/120854051-33a69280-c542-11eb-9a1b-72c3ef85146f.jpg)

**To Reproduce**

I can't share the application but I can provide information as needed.

**Expected behavior**

The build should not fail.

**Please paste the results of `webpack-cli info` here, and mention other relevant information**

webpack: 5.38.1
webpack-cli: 4.7.0
angular: 12.0.3
chrome: 91.0.4472.77
nodejs: 14

**Additional context**

The only similar issue I could find was the following:

[https://github.com/webpack/webpack-cli/issues/1919](url)

**Comment by alexander-akait** [comment]
> I can't share the application but I can provide information as needed.

Sorry in this case we can't help you, nobody can help

**Comment by angulardev123** [comment]
I fixed the deprecation warning and added code to log the size of each string in the Items array in line 791.  I tried to output the contents of each string but some were extremely large.  Below is the length of each item:

```
Item [0] length: 311
Item [1] length: 454
Item [2] length: 107
Item [3] length: 1353
Item [4] length: 110
Item [5] length: 184
Item [6] length: 956
Item [7] length: 224
Item [8] length: 387
Item [9] length: 1031
Item [0] length: 6973216
Item [1] length: 6986403
Item [2] length: 7019187
Item [3] length: 7043118
Item [4] length: 7073040
Item [5] length: 7076430
Item [6] length: 7098872
Item [7] length: 7114246
Item [8] length: 7148054
Item [9] length: 7161969
Item [10] length: 7174903
Item [11] length: 7180068
Item [12] length: 7211812
Item [13] length: 7346789
Item [14] length: 7352609
Item [15] length: 7508869
Item [16] length: 7542899
Item [17] length: 7597820
Item [18] length: 7643825
Item [19] length: 7696057
Item [20] length: 7708144
Item [21] length: 7765180
Item [22] length: 7781531
Item [23] length: 7843863
Item [24] length: 7857128
Item [25] length: 7858693
Item [26] length: 7868910
Item [27] length: 7884152
Item [28] length: 7941229
Item [29] length: 9659558
Item [30] length: 9664369
Item [31] length: 10037578
Item [32] length: 10116817
Item [33] length: 10128905
Item [34] length: 10134749
Item [35] length: 10137503
Item [36] length: 10144837
Item [37] length: 10168465
Item [38] length: 10169834
Item [39] length: 10173749
Item [40] length: 10178032
Item [41] length: 10181067
Item [42] length: 10188074
Item [43] length: 10245104
Item [44] length: 10263826
Item [45] length: 10301087
Item [46] length: 10305660
Item [47] length: 10311253
Item [48] length: 10324736
Item [49] length: 10342761
Item [50] length: 10346614
Item [51] length: 10349790
Item [52] length: 10364112
Item [53] length: 10371419
Item [54] length: 10375264
Item [55] length: 10388727
Item [56] length: 10390396
Item [57] length: 10394918
Item [58] length: 10404431
Item [59] length: 10443795
Item [60] length: 10509331
Item [61] length: 10531931
Item [62] length: 10552180
Item [63] length: 10577049
Item [64] length: 10626793
Item [65] length: 10627722
Item [66] length: 10647428
Item [67] length: 10888294
Item [68] length: 10955194
Item [69] length: 11019649
Item [70] length: 11057925
Item [71] length: 11084667
Item [72] length: 11093268
Item [73] length: 11093780
Item [74] length: 11096279
Item [75] length: 11101267
Item [76] length: 11105858
Item [77] length: 11111312
Item [78] length: 11118338
Item [79] length: 11119159
Item [80] length: 11119779
Item [81] length: 11127946
Item [82] length: 11128798
Item [83] length: 11131359
Item [84] length: 11132196
Item [85] length: 11132967
Item [86] length: 11133807
Item [87] length: 11137659
Item [88] length: 11157549
Item [89] length: 11158046
Item [90] length: 11166405
Item [91] length: 11182764
Item [92] length: 11205823
Item [93] length: 12082586
Item [94] length: 12090419
Item [95] length: 12102483
Item [96] length: 12103789
Item [97] length: 12115165
Item [98] length: 12117143
Item [99] length: 12128334
Item [100] length: 12143673
Item [101] length: 12153938
Item [102] length: 12187726
Item [103] length: 12296913
Item [104] length: 12320449
Item [105] length: 12359856
Item [106] length: 12360018
Item [107] length: 12361136
Item [108] length: 12370939
Item [109] length: 12422833
Item [110] length: 12437135
Item [111] length: 12998288
Item [112] length: 13070672
Item [113] length: 13323914
Item [114] length: 13340851
Item [115] length: 13372235
Item [116] length: 13692844
Item [117] length: 13701472
Item [118] length: 13896738
Item [119] length: 13898553
Item [120] length: 9591
Item [121] length: 4667
Item [122] length: 2301
Item [123] length: 709
Item [124] length: 2920
Item [125] length: 3013
Item [126] length: 2018
Item [127] length: 2970
Item [128] length: 2303
Item [129] length: 3066
Item [130] length: 4379
Item [131] length: 5097
Item [132] length: 5052
Item [133] length: 5708
Item [134] length: 6364
Item [135] length: 223948
Item [136] length: 226327
Item [137] length: 2480
Item [138] length: 1377
Item [139] length: 1654
Item [140] length: 3452
Item [141] length: 1139
Item [142] length: 1371
Item [143] length: 1625
Item [144] length: 1022806
Item [145] length: 1024048
Item [146] length: 1029159
Item [147] length: 1032488
Item [148] length: 1038345
Item [149] length: 1040834
Item [150] length: 1061571
Item [151] length: 1066485
Item [152] length: 1160366
Item [153] length: 1211180
Item [154] length: 1212750
Item [155] length: 1485618
Item [156] length: 1569959
Item [157] length: 1642915
Item [158] length: 321366
Item [159] length: 328037
Item [160] length: 340072
Item [161] length: 364110
Item [162] length: 456991
Item [163] length: 526096
Item [164] length: 532578
Item [165] length: 537900
Item [166] length: 557957
Item [167] length: 113965
Item [168] length: 114531
Item [169] length: 115062
Item [170] length: 118208
Item [171] length: 184500
Item [172] length: 10880
Item [173] length: 6945
Item [174] length: 8130
Item [175] length: 3565
Item [176] length: 2525
Item [177] length: 1690
Item [178] length: 2018
Item [179] length: 1402
Item [180] length: 58687
Item [181] length: 64051
Item [182] length: 87930
Item [183] length: 271011
Item [184] length: 1039457
Item [185] length: 1240456
Item [186] length: 1263672
Item [187] length: 1272601
Item [188] length: 1289501
Item [189] length: 1353394
Item [190] length: 1415059
Item [191] length: 1446440
Item [192] length: 1460453
Item [193] length: 1485767
Item [194] length: 1489946
Item [195] length: 1495065
Item [196] length: 1570297
Item [197] length: 1839742
Item [198] length: 1855179
Item [199] length: 2284497
Item [200] length: 2310659
Item [201] length: 2325911
Item [202] length: 2360916
Item [203] length: 2374917
Item [204] length: 2391039
Item [205] length: 2421528
Item [206] length: 2434846
Item [207] length: 433701
Item [208] length: 459187
Item [209] length: 481136
Item [210] length: 610350
Item [211] length: 665038
Item [212] length: 682106
Item [213] length: 736423
Item [214] length: 744849
Item [215] length: 749401
Item [216] length: 759567
Item [217] length: 778079
Item [218] length: 818883
Item [219] length: 2896094
Item [220] length: 2906726
Item [221] length: 2916128
Item [222] length: 2931071
Item [223] length: 2991699
Item [224] length: 3014570
Item [225] length: 3040178
Item [226] length: 3096305
Item [227] length: 3183169
Item [228] length: 3185823
Item [229] length: 3188451
Item [230] length: 3191122
Item [231] length: 3233853
Item [232] length: 3249282
Item [233] length: 3286796
Item [234] length: 3303748
Item [235] length: 3324978
Item [236] length: 3403309
Item [237] length: 3423232
Item [238] length: 3465073
Item [239] length: 3504474
Item [240] length: 3561434
Item [241] length: 3615149
Item [242] length: 3634040
Item [243] length: 3678446
Item [244] length: 3693830
Item [245] length: 3710562
Item [246] length: 3719000
Item [247] length: 3730163
Item [248] length: 3766297
Item [249] length: 3940392
Item [250] length: 4009983
Item [251] length: 4030095
Item [252] length: 4064304
Item [253] length: 4126108
Item [254] length: 4156191
Item [255] length: 4211811
Item [256] length: 4277202
Item [257] length: 4628827
Item [258] length: 4650132
Item [259] length: 4671906
Item [260] length: 4680155
Item [261] length: 4699938
Item [262] length: 4936934
Item [263] length: 4972903
Item [264] length: 4994523
Item [265] length: 5015836
Item [266] length: 5052536
Item [267] length: 5107242
Item [268] length: 5189159
Item [269] length: 5206227
Item [270] length: 5232259
Item [271] length: 5248554
Item [272] length: 5309099
Item [273] length: 5374601
Item [274] length: 5378376
Item [275] length: 5419598
Item [276] length: 5426382
Item [277] length: 5438900
Item [278] length: 5442231
Item [279] length: 5473018
Item [280] length: 5571775
Item [281] length: 5645437
Item [282] length: 5706001
Item [283] length: 5741742
Item [284] length: 1660429
Item [285] length: 1706810
Item [286] length: 1861857
Item [287] length: 1901445
Item [288] length: 1924926
Item [289] length: 1962926
Item [290] length: 1986499
Item [291] length: 2003604
Item [292] length: 2029755
Item [293] length: 2037791
Item [294] length: 2048998
Item [295] length: 2054120
Item [296] length: 2089808
Item [297] length: 2183459
Item [298] length: 2629960
Item [299] length: 2659288
Item [300] length: 2671917
Item [301] length: 2681102
Item [302] length: 2690323
Item [303] length: 2700499
Item [304] length: 2711881
Item [305] length: 2719890
Item [306] length: 2721818
Item [307] length: 2751226
Item [308] length: 2852758
Item [309] length: 3066652
Item [310] length: 3199689
Item [311] length: 844518
Item [312] length: 850136
Item [313] length: 905824
Item [314] length: 959770
Item [315] length: 969293
Item [316] length: 1020655
Item [317] length: 1071686
Item [318] length: 1139939
Item [319] length: 1140822
Item [320] length: 1298127
Item [321] length: 1350921
Item [322] length: 1385758
Item [323] length: 1420435
Item [324] length: 1421842
Item [325] length: 1579311
Item [326] length: 1635069
Item [327] length: 1673584
Item [328] length: 717895
Item [329] length: 726653
Item [330] length: 736825
Item [331] length: 743180
Item [332] length: 751992
Item [333] length: 758589
Item [334] length: 768510
Item [335] length: 780578
Item [336] length: 793993
Item [337] length: 802708
Item [338] length: 817136
Item [339] length: 828536
Item [340] length: 842961
Item [341] length: 857322
Item [342] length: 868368
Item [343] length: 873516
Item [344] length: 886768
Item [345] length: 895608
Item [346] length: 956207
Item [347] length: 1006766
Item [348] length: 1080677
Item [349] length: 1087161
Item [350] length: 1101476
Item [351] length: 1113465
Item [352] length: 1117249
Item [353] length: 1121849
Item [354] length: 1123053
Item [355] length: 1160666
Item [356] length: 1196088
Item [357] length: 1219068
Item [358] length: 1227800
Item [359] length: 1258360
Item [360] length: 1297709
Item [361] length: 1383681
Item [362] length: 1390957
```

**Comment by angulardev123** [comment]
I figured out what the problem was.  As I was testing different webpack configurations, I renamed some the dist output folders of some of the configurations and left them inside the application folder (i.e. dist - webpack test 1, etc.).  Since they were left in the application folder they were being processed by webpack in subsequent builds.  I noticed this by analyzing the output of each item in the Items array.  As soon as I removed these dist folders and placed them somewhere else the problem went away.  Thanks for the help.
